### PR TITLE
Added strict mode for proprogating failed SSH into failed cast call

### PR DESCRIPTION
**Pull request by smreichling**
- [x] reviewed by @bakks

### CODE REVIEW

File: lib/cast.rb [hunk 1]
```diff
@@ -51,24 +51,32 @@ def self.expand_groups cmdgroups, groups = @@groups
...
 
-  def self.run hosts, cmd, serial = false, delay = nil, ssh = 'ssh'
+  def self.run hosts, cmd, serial = false, delay = nil, ssh = 'ssh', strict = false
+    success = true
     if serial or delay
       hosts.each_with_index do |host, i|
-        remote host, cmd, ssh
+        exitCode = remote host, cmd, ssh
+        if strict
+          success &= (exitCode == 0)
```
Comment (bakks) [inline lib/cast.rb]: I bet this works but using `&&=` here would be a little clearer.

**Comment by bakks** [comment]
:+1:

File: lib/cast.rb (hunk 1 above)
Comment (smreichling) [inline lib/cast.rb]: Interesting, didn't know ruby had that. Fixed.

**Comment by smreichling** [comment]
I think you have to merge Peter, it says I don't have write access.

**Pull request merged**